src/main/java/org/folio/services/finance/transaction/PendingPaymentWorkflowService.java

```diff
@@ -137,11 +137,14 @@ private List<InvoiceWorkflowDataHolder> withNewPendingPayments(List<InvoiceWorkf
   }
 
   private Future<Void> updateTransactions(List<InvoiceWorkflowDataHolder> holders, RequestContext requestContext) {
-    var futures = holders.stream()
-      .map(InvoiceWorkflowDataHolder::getNewTransaction)
-      .map(transaction -> baseTransactionService.updateTransaction(transaction, requestContext))
-      .collect(Collectors.toList());
-    return GenericCompositeFuture.join(futures).mapEmpty();
+    for (InvoiceWorkflowDataHolder holder : holders) {
+      Transaction pendingPayment = holder.getNewTransaction();
+      baseTransactionService.updateTransaction(pendingPayment, requestContext).recover(t -> {
+          logger.error("Failed to update pending payment with id {}", pendingPayment.getId(), t);
+          throw new HttpException(500, PENDING_PAYMENT_UPDATE_ERROR.toError());
+        });
+    }
+    return Future.succeededFuture();
```

Comment:
All the logic above is not connected with the method result, so this method will always return `Future.succeededFuture()`. I suppose exception PENDING_PAYMENT_UPDATE_ERROR could be seen in logs, but it doesnt affect the workflow at all. I dont think this is expected behavior

Reply:
Please have a look how sequential processing implemented here https://github.com/folio-org/mod-invoice/blob/master/src/main/java/org/folio/services/finance/transaction/PaymentCreditWorkflowService.java#L78

Reply:
I suppose to use [chainCall method](https://github.com/folio-org/mod-finance-storage/blob/master/src/main/java/org/folio/rest/persist/HelperUtils.java#L158). You can find an example by this [link](https://github.com/folio-org/mod-finance-storage/blob/master/src/main/java/org/folio/service/rollover/RolloverBudgetService.java#L63)

Reply:
@grabsefx , @Abdulkhakimov , @SerhiiNosko 
Cant we use semaphore here as below
https://github.com/folio-org/mod-invoice/blob/master/src/main/java/org/folio/services/finance/transaction/PendingPaymentWorkflowService.java#:~:text=%7D-,private%20Future%3CVoid%3E%20createPendingPayments(List%3CInvoiceWorkflowDataHolder%3E,%7D,-private%20Future%3C

Reply:
@SinghAdes I see you used approach with semaphores to do parallel processing in 5 active threads, so it is not sequnetial processing one by one that @grabsefx and @Abdulkhakimov proposed, is it working for you?

Reply:
And basically it's hard to determine is it working or not with semaphores because it is multitreading execution and it can work now but it will not for the next run.
So to be on the safe side I think better to do sequential execution without semaphores until we will refactor AllOrNothing approach.

Reply:
@SerhiiNosko , @grabsefx , @Abdulkhakimov :- I tried with the solution suggested above but none of them worked . if we are returning future its still failing with "PRIMARY KEY VOILATION". 
Anyway i tried 2 operations **(update summary & update transaction)** run  separately instead of making them dependent on each other and this works . Please review.

Reply:
I could not understand how this fix can help resolve this issue, now these 2 operations are not sequential, so we need to update invoice summary fistly and that updating transactions